firewood/src/v2/api.rs

```diff
@@ -154,6 +154,10 @@ pub enum Error {
     /// Revision not found
     #[error("revision not found")]
     RevisionNotFound,
+
+    /// An invalid root hash was provided
+    #[error("provided root hash is not a valid hash key")]
```

Comment:
nit: how is a root hash provided? In the eyes of any user, the root hash is retrieved, and only exposed through the function that returns this error.
```suggestion
    #[error("root hash found is not a valid hash key")]
```